src/server/src/routes/workExperience/index.ts

```diff
@@ -0,0 +1,177 @@
+import StatusCodes from 'http-status-codes';
+import { Request, Response, Router } from 'express';
+import { WorkExperience } from '@entities/workExperience';
+import { errors } from '@shared/errors';
+import passport from 'passport';
+import {
+    createWorkExperience,
+    getWorkExperiences,
+    updateWorkExperience,
+    deleteWorkExperience
+} from '@modules/workExperience';
+import logger from '@shared/Logger';
+import { validationMiddleware } from '@middlewares/validation';
+import { 
+    workExperienceCreateSchema,
+    workExperienceReadSchema,
+    workExperienceUpdateSchema,
+         } from './schemas';
+
+const router = Router();
+const {
+    CREATED,
+    OK,
+    INTERNAL_SERVER_ERROR,
+    UNAUTHORIZED,
+    BAD_REQUEST,
+} = StatusCodes;
+
+interface workExperienceRequest extends Request {
+    body: {
+        workExperience: WorkExperience;
+    };
+}
+
+interface workExperienceIdRequest extends Request {
+    body: {
+        workExperienceId: number;
+    };
+}
+/******************************************************************************
+ *            POST Request - Create - /api/workExperience/create
+ ******************************************************************************/
+
+router.post(
+    '/create',
+    passport.authenticate('jwt', { session: false }),
+    validationMiddleware({ bodySchema: workExperienceCreateSchema }),
+    async (req: workExperienceRequest, res: Response) => {
+        const {
+            startDate,
+            endDate, 
+            description,
+            employer,
+            title,
+            studentId,
+        } = req.body.workExperience;
+
+        try {
+            const { result, message } = await createWorkExperience(
+                description,
+                employer,
+                startDate,
+                endDate, 
+                title,
+                studentId,
+            );
+            return result
+                ? res.status(CREATED).end()
+                : res.status(BAD_REQUEST).json({ error: message }).end();
+        } catch (error) {
+            logger.err(error);
+            return res
+                .status(INTERNAL_SERVER_ERROR)
+                .json(errors.internalServerError)
+                .end();
+        }
+    }
+);
+
+/******************************************************************************
+ *            GET Request - Read - /api/workExperience/read
+ ******************************************************************************/
+
+interface WorkExperienceReadRequest extends Request {
+    query: {
+        title: string;
+        startDate: string;
+        endDate: string;
+        employer: string; 
+        description: string; 
+        numOfItems: string;
+    };
+}
+
+router.get(
+    '/read',
```

Comment:
This route should only get the studentId. For example in the params.

`/read/:studentId`

Then you can get the studentId like this:

`const { studentId} = req.params`

Then you can pass that studentId to your getWorkExperiences helper method. I don't think you need none of the other data you have in WorkExperienceReadRequest.

Reply:
You forgot to update the url of this endpoint. It should be:

`router.get(
    '/read/:studentId' ....`